### PR TITLE
Localize required fields note on application summary view and make the * consistently red

### LINKED ISSUE #5536: Required fields indicator text is not localized in application summary view
**Describe the bug**
The required fields explanation text ("Note: Fields marked with a * are required.") is not localized on the application summary page:

Also, the asterisk in the explanation text is not red in the application block view.  See the TI dashboard for the desired output.  

**To Reproduce**

1. From the landing page, select a program and click "Apply"
2. Select a language other than English from the language dropdown.
3. Note that "Note: Fields marked with a * are required." is not translated.

**Expected behavior**
The required fields indicator text should be localized on the application summary page.
The asterisk in the required fields text should be red in the application blocks view.

**Screenshots**
image

image

**Additional context**
The text has been localized in preparation for this work as `content.requiredFieldsAnnotationFirstHalf` and `content.requiredFieldsAnnotationSecondHalf` as part of this [PR](https://github.com/civiform/civiform/pull/5496)

### PR DESCRIPTION
**Pull request by melanie-exygy**
### Description

Localized the required fields explanation text ("Note: Fields marked with a * are required.") on the application summary page.

Also, made the asterisk red in the explanation text of the application block view.  This means the * will be consistently red across the application.  

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Instructions for manual testing

1. From the landing page, select a program and click "Apply".
2. Select a language other than English from the language dropdown.
3. Note that the required fields note is translated.
4. Click "Continue".
5. Note that the * is red.

image

### Issue(s) this completes

Fixes #5536

### CODE REVIEW

File: server/app/views/applicant/TrustedIntermediaryDashboardView.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public Content render(
                 h2(tiGroup.getDescription()).withClasses("ml-2"),
                 hr(),
                 renderHeader("Add Client"),
-                BaseHtmlView.requiredFieldsExplanationContent(),
+                requiredFieldsExplanationContent(),
```
Comment (melanie-exygy) [inline server/app/views/applicant/TrustedIntermediaryDashboardView.java:81]: Given that `TrustedIntermediaryDashboardView` extends `BaseHtmlView`, it isn't necessary to specify `BaseHtmlView` here.

File: server/app/views/ApplicationBaseView.java [hunk 2]
```diff
@@ -136,8 +139,9 @@ public abstract Builder setErrorDisplayMode(
    * @param messages the localized {@link Messages} for the current applicant
    * @return PTag containing requiredness text.
    */
-  public PTag requiredFieldsExplanationContent(Messages messages) {
-    return p(messages.at(MessageKey.REQUIRED_FIELDS_ANNOTATION.getKeyName()))
+  public static PTag requiredFieldsExplanationContent(Messages messages) {
+    SpanTag redAsterisk = span("*").withClass("text-red-700");
```
Comment (dkatzz) [inline server/app/views/ApplicationBaseView.java:143]: **thought:** If this is a color we're going to start using more consistently, we can add it to the base styles with a constant name https://github.com/civiform/civiform/blob/f1f5a42151b63aee20210ddfda4a3778dd24c82e/server/app/views/style/BaseStyles.java#L45

**Review by dkatzz — APPROVED** [review]

File: server/app/views/ApplicationBaseView.java (hunk 2 above)
Comment (melanie-exygy) [reply server/app/views/ApplicationBaseView.java:143]: That's a good idea.  I'll add that style.

Comment (melanie-exygy) [reply server/app/views/ApplicationBaseView.java:143]: Realized that the other red asterisks on the page were "text-red-600", so, for consistency, I'm going to use the already existing BaseStyle.FORM_ERROR_TEXT_COLOR that you mentioned.

**Pull request merged**